### PR TITLE
Inline argument conditionals

### PR DESCRIPTION
**Pull request by niik**
## Description

While investigating an unrelated issue I had to figure out what arguments were being passed to Git and found our current approach of nested conditionals to be difficult to read. This PR simplifies (I hope) the code by using inline conditionals to make it easier to understand and (hopefully) maintain going forward.

Maybe it's just me though and this isn't all that much cleaner? Thoughts?

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/lib/git/core.ts [hunk 1]
```diff
@@ -485,8 +485,7 @@ export function gitRebaseArguments() {
     // uses the merge backend even if the user has the apply backend
     // configured, since this is the only one supported.
     // This can go away once git deprecates the apply backend.
-    '-c',
-    'rebase.backend=merge',
+    ...['-c', 'rebase.backend=merge'],
```
Comment (sergiou87) [inline app/src/lib/git/core.ts:488]: This change feels excessive since there are no conditions 🤔

Comment (niik) [reply app/src/lib/git/core.ts:488]: So these ones I really like actually, we use them here and there like in https://github.com/desktop/desktop/blob/f921b4282c8b4c8c0c3c63bf52b4b8f81e3a8ce2/app/src/lib/is-application-bundle.ts#L35-L36 as a way to illustrate that they are semantically grouped. If you remove one you got to remove the other.

S--8HlEg

Comment (sergiou87) [reply app/src/lib/git/core.ts:488]: > as a way to illustrate that they are semantically grouped

Ohhhhhh :sold: :shipit:

**Review by sergiou87 — APPROVED** [review]

**Review by tidy-dev — APPROVED** [review]
💖  Is easier to digest.

And, this makes me think of having `classNames` like function where you can feed it a `{ '--progress': progressCallback}` and it adds the args as appropriate.

```
args('arg1', `arg2`, {'arg3': arg3condition, `arg4`: arg4condition})
```

**Comment by niik** [comment]
> And, this makes me think of having classNames like function

Awww yeah. I tested this out real quick in https://github.com/desktop/desktop/commit/c36ad79ecb098e3b93885acadd0ba1653e657639. I won't have time to wrap that up before my leave but I think it has promise

**Pull request merged**